src/components/Farmhand/FarmhandReducers.js

```diff
@@ -0,0 +1,227 @@
+/** @typedef {typeof import('./Farmhand').default.defaultProps} farmhand.props */
+/** @typedef {import('./Farmhand').farmhand.state} farmhand.state */
+import { Component } from 'react'
+
+import * as reducers from '../../game-logic/reducers'
+
+/**
+ * @callback BoundReducer
+ * @param {...any} rest
+ * @returns {farmhand.state}
+ */
+
+/**
+ * This class serves as a sort of middleware for the main Farmhand component.
+ * It defines stub methods with names that correspond to various reducer
+ * functions. At the time of instantiation, it implements the methods with the
+ * matching reducer function and binds it to the class instance.
+ *
+ * This is done in the interest of coding convenience and decentralization of
+ * core game logic. It is implemented as a class so that the methods can be
+ * statically analyzed by the TypeScript compiler.
+ */
+export class FarmhandReducers extends Component {
```

Comment:
The goal with this weird pattern in to replace the old `initReducers` method with an alternative solution that can be statically analyzed: https://github.com/jeremyckahn/farmhand/blob/ea69b10b6338db9a272fd6db7e622c94e6033486/src/components/Farmhand/Farmhand.js#L544-L604

With this, we're able to still maintain just a single list of functions to be bound without redundant pass-through logic for each one (see the `constructor` below).

Reply:
i really don't like inheritance when it comes to react components, but i do think what you have here is an improvement vs the way it used to be (which was a little too "magic" feeling for my tastes). i don't have a better suggestion off hand so i'm generally cool with what you've cooked up here, but i may think on it some more and see if i can come up with another suggestion

Reply:
Yeah I'm not thrilled about this either for the same reasons. It's the least-bad solution I could come up with that was syntactically valid and type-safe. Let's run with this for now and revisit if something better comes to us.